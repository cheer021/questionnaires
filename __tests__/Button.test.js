import React from 'react';
import SimpleButton from '../src/components/Button';
import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
test('snapshot :: button', () => {
	const component = mount(<SimpleButton disabled/>);
	expect(toJson(component)).toMatchSnapshot();
});


test('onClick: button', ()=> {
	const mockOnClick = jest.fn();
	const component = mount(<SimpleButton onClick ={mockOnClick}/>);
	component.simulate('click');
	expect(mockOnClick.mock.calls.length).toBe(1);
})

