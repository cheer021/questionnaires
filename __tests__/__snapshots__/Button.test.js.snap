// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot :: button 1`] = `
<SimpleButton
  disabled={true}
>
  <Button
    active={false}
    disabled={true}
    type="button"
    variant="primary"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      type="button"
    />
  </Button>
</SimpleButton>
`;
